File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
    if (s.length != t.length) return false;
    let map = {}
    for (let i of s) {
        map[i] = (map[i] || 0) + 1
    }
    for (let i of t) {
        if (map[i] === undefined) {
            return false;
        }
        map[i]--;
    };
    for (let i in map) {
        if (map[i] !== 0) {
            return false;
        }
    }
    return true;
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        freq = {}
        for i in s:
            freq[i] = freq.get(i, 0) + 1
        for i in t:
            if i not in freq:
                return False
            freq[i] -= 1
        for i in freq:
            if freq[i] != 0:
                return False
        return True